docs/getting_started/basics/terminology.md

```diff
@@ -1,21 +1,43 @@
 # Terminology
 
-Here are some of the basic terminology used in Shift:
+This page assembles a list of definitions for some of Shift's main concepts.
 
-## Operator
+## Workflow
 
-An operator is an execution block that is delegated of performing a specific action given a set of input data if required. From a programming perspective an operator can be seen as a function that takes some input data and processes it to produce some output data. Operators in Shift can be connected to propagate information between each other.  
+A workflow is a directed acyclic graph (DAG) that builds the casual relationship between the execution of its nodes. A workflow is composed of interconnected nodes and defines the global execution logic of the automation process.
 
-*\*operator* and *node* will be used interchangeably.  
+## Node
+
+Nodes are the graphic block units that build up the workflow. Each node is associated with an operator, which defines the execution logic of the node and is connected to other nodes through plugs.
 
 ## Plug
 
-A plug is a Shift object containing the data that is passed to or from a certain operator. An operator has input plugs on its left-side and output on its right-side.
+A plug represents an input or output of a node. Plugs have associated types based on the Python object that they should contain. Plug values can be set manually via the inspector widget or propagated from a connected plug.
 
-## Workflow
+## Connection
+
+A connection is the graphical representation of the relationship between two plugs and shows the direction of the data propagation. Connections between plugs from different nodes are what defines the execution stack of the workflow. 
+
+## Execution Stack
+
+The execution stack refers to the order in which nodes will be executed. The execution stack is closely related to the workflow configuration. It can be a linear execution list of nodes in the workflow or dynamically expand or contract under the presence of iterators or conditionals. 
+
+## Board
 
-A Shift workflow is a directed acyclic graph (DAG) that builds the causal relationship between the execution of its nodes.
+The board refers to the workspace in the Shift UI where the graphical counterpart of the workflow is build. It is the master object holding all information on the graphic items in the interface.
```

Comment:
I would change the `master` term to use `main`, as happened to github with the branch naming:
```suggestion
The board refers to the workspace in the Shift UI where the graphical counterpart of the workflow is build. It is the main object holding all information on the graphic items in the interface.
```